storage/sql.go

```diff
@@ -0,0 +1,378 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/go-sql-driver/mysql"
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/telemetry"
+	tupleUtils "github.com/openfga/openfga/pkg/tuple"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	DefaultMaxTuplesPerWrite             = 100
```

Comment:
Should we move these into `storage/storage.go` since they are more generic to the storage interface than the `sql` specific signatures?

Reply:
Ah, yes. Thank you. Done.